### ISSUE TITLE
empress vs. emperor

### ISSUE THREAD

**Issue by ARW-UBT** [biocore/empress#399]
Hi, 
Is it necessary to uninstall emperor before installing empress? 
I am not a plugin developer, just a ordinary user and I am wondering whether uninstalling emperor will affect display of pcoa or rpca plots (or any other plots that use emperor). 
Are empress and emperor not compatible on the same installation (currently q2-2020.8)?

Thanks for the first viewer for trees in q2!

**Comment by kwcantrell** [comment]
Thanks for having an interest in empress! When integrating emperor and empress (which we are calling an empire plot!), we had to update parts of the emperor code base. This means that you will need to have the most up-to-date version of emperor in order to full take advantage of the empire plot. So, to make life easier for the users, we have you uninstall emperor so that empress can then install the most up-to-date version of emperor. Perhaps we should make a note in the installation instructions that lets users know empress will reinstall emperor.

tldr: empress will reinstall the most up-to-date version of emperor.

If you have any addition questions please let us know!

**Comment by ARW-UBT** [comment]
"empress will reinstall emperor"
Thank you, this was not clear to me... :-)
Thank you!

**Comment by ElDeveloper** [comment]
Thanks for checking @ARW-UBT

**Comment by ARW-UBT** [comment]
Great tool! Thanks!
There is only one issue that I could not solve. 
The empress-tree.qzv opens correctly also with my own data if I invoke it on the linux machine via 'qiime tools view ...'
However, the same file does not open, if I load it into the q2 web viewer view.qiime2.org. In particular, the icon for the 'sharable link' and for download are not displayed in the top line next to 'provenance'. See p1.png (my visualization) and p2.png (your sample visualization). 
Is the standard qiime2 web viewer not able to display the qzv?

![p1](https://user-images.githubusercontent.com/16762238/94579878-8de7a900-0279-11eb-9654-98b4416a8ed2.PNG)
![p2](https://user-images.githubusercontent.com/16762238/94579899-9213c680-0279-11eb-8f0f-aa9c8314fede.PNG)

**Comment by ElDeveloper** [comment]
Would you mind trying a different web browser? Or alternatively verifying that you can visualize other QZVs?

**Comment by fedarko** [comment]
Not sure why the file isn't loading, but the two icons ("shareable link" and download) not showing up when you use view.qiime2.org is expected -- I believe they only show up when view.qiime2.org is given a link to a QZV, rather than having a QZV uploaded.

**Comment by ARW-UBT** [comment]
Hello, 
I have tried Firefox, Chrome, and Opera: The same effect, none of these browwser load the qzv visualization via view.qiime2.org. 
This applies to visualizations created in my q2-2020.8 installation for my own data as well as for the example dataset, which is available on github (downloaded the files and created a visualization again). 

However, when I download the qzv from https://mchelper.ucsd.edu/downloads/empress/master/empress-tree.qzv, this tree can be loaded with all available option in Firefox, Chrome, and Opera directly via view.qiime2.org. 

Please tell me if you need more details on my installation.

Best,

**Comment by ARW-UBT** [comment]
Two screenshots of the same qzv, one via view-qiime2.org, the other via qiime tools view...

![empress_q2-tools-view](https://user-images.githubusercontent.com/16762238/94650917-da71c980-02f7-11eb-9fc5-30e703d68826.PNG)
![empress-view-q2-org](https://user-images.githubusercontent.com/16762238/94650924-dcd42380-02f7-11eb-9445-3e263c1d0b8f.PNG)

**Comment by ARW-UBT** [comment]
Additional note: same behaviour on an Ubuntu installation (original image from q2 virtualbox) and on a CentOS7 (own setup).

**Comment by ARW-UBT** [comment]
...Or alternatively verifying that you can visualize other QZVs?

I have tested it with biplots and qurro plots created today with your sample data. Seem to be ok, can be loaded without problems via: 
> qiime tools view ... on the q2 machine
view.qiime2.org on the q2 machine,
view.qiime2.org after transfer to my Windows desktop
using Opera and Firefox.

**Comment by kwcantrell** [comment]
Would you mind sharing the .qzv? I created a fresh environment with qiime2-2020.8 and empress and I am able to load the .qzv in view.qiime2.org.

**Comment by ARW-UBT** [comment]
Yes, of course. These are a visualization that I created 'after' installing empress. The biplot and the qurro plot load just fine via the browser (and via qiime tools view), but the empress-tree.qzv can be loaded only from qiime tools view. 

They were zipped again to drag&drop here...
[Temp.zip](https://github.com/biocore/empress/files/5306831/Temp.zip)

**Comment by kwcantrell** [comment]
you are missing "spectum.min.css", which causes view.qiime2.org to fail. I unzipped the .qzv to look at the code and it seems like you have an older version of empress. Try to reinstall empress.

```
pip uninstall --yes empress
pip uninstall --yes emperor
pip install git+https://github.com/biocore/empress.git
qiime dev refresh-cache
qiime empress --help
```

**Comment by kwcantrell** [comment]
Let me know if that works. After looking closer at our "setup.py", I think you are still going to have an error. Our "setup.py" will not include "spectum.js.css". I have a local copy of the repo so I have "spectum.js.css".

**Comment by fedarko** [comment]
Awesome catch -- yeah, I was super confused because it looks like the version of Empress used to create the artifact is relatively recent. Adding an extra line to `setup.py` that says `support_files/vendor/*.css` should fix this. I think this error is my fault, sorry ._.

https://github.com/biocore/empress/blob/4ab2c8bd5fbd97942b7c42bf928820810714d9da/setup.py#L57-L62

**Comment by kwcantrell** [comment]
@ARW-UBT we just uploaded a fix so you if you reinstall empress again, you should be good to go.

**Comment by ARW-UBT** [comment]
Hi together!
Great, works just fine with remote browsers. 
Thanks for the bug fix. 
Best regards,
